server/app/survey_response.go

```diff
@@ -323,3 +334,56 @@ func (a *UserSurveyApp) hasRatingGroupChanged(oldRating, newRating int) bool {
 		return true
 	}
 }
+
+func (a *UserSurveyApp) HandleRefreshSurveyPost(userID, postID string) error {
+	post, appErr := a.api.GetPost(postID)
+	if appErr != nil {
+		a.api.LogError("HandleRefreshSurveyPost: failed to get post by ID", "postID", postID, "error", appErr.Error())
+		return errors.Wrapf(errors.New(appErr.Error()), "HandleRefreshSurveyPost: failed to get post by ID, postID: %s", postID)
+	}
+
+	if post.Type != surveyPostType {
+		return errors.New("Post is not a survey post")
+	}
+
+	// verify post belongs to the user's DM
+	channel, appErr := a.api.GetChannel(post.ChannelId)
+	if appErr != nil {
+		a.api.LogError("HandleRefreshSurveyPost: failed to channel by ID", "channelID", post.ChannelId, "error", appErr.Error())
+		return errors.Wrapf(errors.New(appErr.Error()), "HandleRefreshSurveyPost: failed to get channel by ID, channelID: %s", post.ChannelId)
+	}
+
+	if !strings.Contains(channel.Name, userID) {
+		return fmt.Errorf("HandleRefreshSurveyPost: post does not belong to the user, postID: %s, userID: %s", postID, userID)
+	}
```

Comment:
We check the channel contains the user, but we don't check the post is created by the bot, do we? There is probably no real way to exploit such vulnerability, so 0/5 on whether to add the check or not.

Related, but even more difficult to exploit, you are checking the channel is with the user, but not necessarily a DM with the bot.

Reply:
I should mention this in code, but this is intentional. Lets say the post was created by the bot, and the bot's ID was, say, `bot_id_1`. If, now, for any reason the bot is deleted, then plugin will create another both with the same name but different ID as IDs are pseudo-random. So, checking for bot ID isn't reliable.